File: app/src/main/java/org/techtown/find_gas_station/ViewModel/SetViewModel.java
package org.techtown.find_gas_station.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import org.techtown.find_gas_station.Repository.SetRepository;
import org.techtown.find_gas_station.set.Set;

public class SetViewModel extends AndroidViewModel {

    private SetRepository setRepository;
    private LiveData<Set> setLiveData;

    public SetViewModel(@NonNull Application application) {
        super(application);
        setRepository = new SetRepository(application);
        setLiveData = setRepository.getAllSets();
    }

    public void insert(Set set) {
        setRepository.insert(set);
    }

    public void update(Set set){
        setRepository.update(set);
    }

    public void delete(){
        setRepository.deleteAll();
    }

    public LiveData<Set> getSetLiveData() {
        return setLiveData;
    }


}

// ViewModel 클래스 내에서 LiveData 정의

File: app/src/main/java/org/techtown/find_gas_station/ViewModel/SetViewModel.kt
package org.techtown.find_gas_station.ViewModel

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import org.techtown.find_gas_station.Data.set.OilData
import org.techtown.find_gas_station.Repository.SetRepository
import org.techtown.find_gas_station.Data.set.RoomDB

class SetViewModel(application: Application) : AndroidViewModel(application) {

    private val _oilLocalData = MutableLiveData<OilData>()
    val oilLocalData : LiveData<OilData> get() = _oilLocalData
    private val setRepository: SetRepository

    init {
        val oilDao = RoomDB.getAppDatabase(application).setDao()
        setRepository = SetRepository(oilDao)
    }

    fun getOilLocalData(){

        viewModelScope.launch(Dispatchers.IO){

            val localData = setRepository.getOilLocalData()

            withContext(Dispatchers.Main){
                _oilLocalData.value = localData
            }

        }

    }
    fun updateData(set: OilData) {

        viewModelScope.launch(Dispatchers.IO){
            setRepository.deleteAll()
            setRepository.insert(set)
        }

    }


    class Factory(val application: Application) : ViewModelProvider.Factory {
        override fun <T : ViewModel> create(modelClass: Class<T>) = SetViewModel(application) as T
    }
}
